tasks/BasePIDController.cpp

```diff
@@ -74,10 +74,26 @@ std::pair<base::LinearAngular6DCommand, LinearAngular6DPIDState> BasePIDControll
         // Angular case
         if (!base::isUnset(reference.angular(i)))
         {
-            command_out.angular[i] =
-                pid.angular[i].update(current_state.angular(i),
-                                   reference.angular(i),
-                                   current_state.time.toSeconds());
+            if (_position_in_world_frame)
```

Comment:
:+1: BasePID is really only meant to run a PID ... shouldn't know about angular vs. linear. The change in current_state could be done the same way in PIDController.

Reply:
My intention first was to do it directly in the `PIDController`, but the problem is that the reference (or `merged_command`) is only read by the `BasePIDController`, namely in the `calcOutput` method. So basically there is no access to the reference in the update hook of `PIDController`, since `PIDControllerBase::updateHook()` is called only in the last line of it which itself calls `BasePIDControllerBase::updateHook()` and then `calcOutput` is called and thus the reference is read.

Reply:
I can still use `base::Angle` to normalize the difference then add it to the current_state, but I am not sure if moving this to the child class is not really possible with the way this task is implemented.